### PR TITLE
Foundations Phoenix

### PR DESCRIPTION
**Pull request by SanGeeky**
- Routing
- Plug
- Controllers
- Views/Templates
- Channels/Presence
- Ecto

### CODE REVIEW

File: lib/hello_web/views/page_view.ex [hunk 1]
```diff
@@ -1,3 +1,25 @@
 defmodule HelloWeb.PageView do
   use HelloWeb, :view
```
Comment (juanhurtado10) [inline lib/hello_web/views/page_view.ex:2]: @SanGeeky what are `View`  does?

**Review by juanhurtado10 — APPROVED** [review]

File: lib/hello_web/router.ex [hunk 2]
```diff
@@ -2,21 +2,69 @@ defmodule HelloWeb.Router do
...
   scope "/", HelloWeb do
     pipe_through :browser
 
     get "/", PageController, :index
+    # get "/redirect_test", PageController, :redirect_test
+    get "/test", PageController, :test
+
+    get "/hello", HelloController, :index
+    get "/hello/:messenger", HelloController, :show
+    resources "/users", UserController
```
[the remark this answers is not in the document]
Comment (SanGeeky) [reply lib/hello_web/router.ex:32]: `resources` create restful endpoints for a controller with their respective actions and methods, similar to RoR

File: lib/hello_web/views/page_view.ex (hunk 1 above)
Comment (SanGeeky) [reply lib/hello_web/views/page_view.ex:2]: View basically is a module that renders and selects the correct template related to a controller, we can define functions here to be used in templates.

File: lib/hello_web/endpoint.ex [hunk 3]
```diff
@@ -50,5 +50,16 @@ defmodule HelloWeb.Endpoint do
   plug Plug.MethodOverride
   plug Plug.Head
   plug Plug.Session, @session_options
+  plug :introspect
```
[the remark this answers is not in the document]
Comment (SanGeeky) [reply lib/hello_web/endpoint.ex:53]: Plug is a kind of middleware that operates in the all request lifecycle and is very useful to handle the behavior of each request.

**Pull request merged**